src/main/java/io/supertokens/inmemorydb/Start.java

```diff
@@ -3007,4 +3008,23 @@ public int countUsersThatHaveMoreThanOneLoginMethodOrTOTPEnabledAndActiveSince(A
             throw new StorageQueryException(e);
         }
     }
+
+    @Override
+    public boolean doesClientIdExistForThisApp(AppIdentifier appIdentifier, String clientId)
+            throws StorageQueryException {
+        try {
+            return OAuthQueries.isClientIdForAppId(this, clientId, appIdentifier);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public void addClientForApp(AppIdentifier appIdentifier, String clientId) throws StorageQueryException {
```

Comment:
```suggestion
    public void addClientForApp(AppIdentifier appIdentifier, String clientId) throws StorageQueryException, ClientAlreadyExistsException {
```

Reply:
Since this is an insert query, it can fail if the client already exists, in this case, we need to communicate that to the core logic, which can be communicated via this exception. If you see other queries, the way we detect this is by parsing the error string thrown from the db query based on the primary key

Reply:
Thanks! Done